constants.h

```diff
@@ -45,9 +48,9 @@ struct videoStream {
 	char subs[2][MAX_CHAR];
 	int subPos;
 	int fpPos;
-	volatile int bufferLength;
-	volatile int semaphore;
-	volatile bool bufferEnd;
+	atomic_int bufferLength;
+	pthread_mutex_t mutex;
+	atomic_bool bufferEnd;

```

Comment:
This is very important.

The length was definitely wrong in the different thread, while the count is used to determine whether read or write actions should be taken.
The end is written and read by both the read and write threads so to guarantee that it's read properly its been made atomic
The semaphore did not work. This mutex does.